middleware/locale/method/header.go

```diff
@@ -0,0 +1,76 @@
+package method
+
+import (
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func detectByHeader(r *http.Request, _ *Settings) []string {
+	browserLocales := parseBrowserLocales(r.Header.Get("Accept-Language"))
+	sort.Sort(browserLocales)
+	return browserLocales.locales()
+}
+
+type browserLocales []*browserLocale
+
+func parseBrowserLocales(text string) browserLocales {
+	browserLocales := browserLocales{}
+
+	for _, part := range strings.Split(text, ",") {
+		part := strings.TrimSpace(part)
+		if part == "" {
+			continue
+		}
+		browserLocales = append(browserLocales, parseBrowserLocale(part))
+	}
+
+	return browserLocales
+}
+
+func (bl browserLocales) Len() int {
+	return len(bl)
+}
+
+func (bl browserLocales) Less(i, j int) bool {
+	return bl[i].q > bl[j].q
+}
+
+func (bl browserLocales) Swap(i, j int) {
+	bl[i], bl[j] = bl[j], bl[i]
+}
+
+func (bl browserLocales) locales() []string {
+	result := make([]string, len(bl))
+	for index, browserLocale := range bl {
+		result[index] = browserLocale.locale
+	}
+	return result
+}
+
+type browserLocale struct {
+	locale string
+	q      float64
+}
+
+func parseBrowserLocale(text string) *browserLocale {
+	parts := strings.Split(text, ";")
+
+	browserLocale := &browserLocale{
+		locale: parts[0],
+		q:      1.0,
+	}
+	if len(parts) > 1 {
+		if values, err := url.ParseQuery(parts[1]); err == nil {
```

Comment:
Why? If it's not ```q=``` followed by something that can be parsed as *uint* or *float* then ignore it anyway.